// @elastic/eui/lib/services/color/manipulation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.lightness = exports.desaturate = exports.saturate = exports.shade = exports.tint = exports.transparentize = void 0;

var _chromaJs = _interopRequireDefault(require("chroma-js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

/**
 * Makes a color more transparent.
 * @param color - Color to manipulate
 * @param alpha - alpha channel value. From 0-1.
 */
var transparentize = function transparentize(color, alpha) {
  return (0, _chromaJs.default)(color).alpha(alpha).css();
};
/**
 * Mixes a provided color with white.
 * @param color - Color to mix with white
 * @param ratio - Mix weight. From 0-1. Larger value indicates more white.
 */


exports.transparentize = transparentize;

var tint = function tint(color, ratio) {
  return _chromaJs.default.mix(color, '#fff', ratio, 'rgb').css();
};
/**
 * Mixes a provided color with black.
 * @param color - Color to mix with black
 * @param ratio - Mix weight. From 0-1. Larger value indicates more black.
 */


exports.tint = tint;

var shade = function shade(color, ratio) {
  return _chromaJs.default.mix(color, '#000', ratio, 'rgb').css();
};
/**
 * Increases the saturation of a color by manipulating the hsl saturation.
 * @param color - Color to manipulate
 * @param amount - Amount to change in absolute terms. 0-1.
 */


exports.shade = shade;

var saturate = function saturate(color, amount) {
  return (0, _chromaJs.default)(color).set('hsl.s', "+".concat(amount)).css();
};
/**
 * Decreases the saturation of a color by manipulating the hsl saturation.
 * @param color - Color to manipulate
 * @param amount - Amount to change in absolute terms. 0-1.
 */


exports.saturate = saturate;

var desaturate = function desaturate(color, amount) {
  return (0, _chromaJs.default)(color).set('hsl.s', "-".concat(amount)).css();
};
/**
 * Returns the lightness value of a color. 0-100
 * @param color
 */


exports.desaturate = desaturate;

var lightness = function lightness(color) {
  return (0, _chromaJs.default)(color).get('hsl.l') * 100;
};

exports.lightness = lightness;
